rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/Diagnostics.java

```diff
@@ -71,9 +80,64 @@ public static <K, V> Map<K, List<V>> groupByKey(Stream<Entry<K, V>> diagnostics)
     }
 
     public static Diagnostic translateDiagnostic(ParseError e, ColumnMaps cm) {
-        return new Diagnostic(toRange(e, cm), e.getMessage(), DiagnosticSeverity.Error, "parser");
+        var message = e.getMessage() + " (irrecoverable)";
```

Comment:
for DSLs, we don't know if error recovery is enabled, so this extra message part is not accurate.